### PR TITLE
improved handling of the input height

### PR DESCRIPTION
**Pull request by Benjamindodgson**
This helps fix some UI/UX with the input height in various scenarios.

### CODE REVIEW

File: iOS/Flows/Conversation/Conversation Detail/ConversationDetailCoordinator.swift [hunk 1]
```diff
@@ -124,7 +124,7 @@ class ConversationDetailCoordinator: PresentableCoordinator<DetailCoordinatorRes
         
         let alertController = UIAlertController(title: title, message: message, preferredStyle: .actionSheet)
         
-        let primaryAction = UIAlertAction(title: "Confirm", style: style, handler: {
+        let primaryAction = UIAlertAction(title: title, style: style, handler: {
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Conversation Detail/ConversationDetailCoordinator.swift:127]: More clearerrrrrr

File: iOS/Flows/Conversation/Input/InputHandlerCoordinator.swift [hunk 2]
```diff
@@ -103,6 +103,7 @@ class InputHandlerCoordinator<Result>: PresentableCoordinator<Result>,
             Task.onMainActorAsync {
                 guard let kind = try? await AttachmentsManager.shared.getMessageKind(for: first, body: text) else { return }
                 self.inputHandlerViewController.swipeableVC.currentMessageKind = kind
+                self.inputHandlerViewController.swipeableVC.inputState = .collapsed
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Input/InputHandlerCoordinator.swift:106]: This feels right to me, would be weird to go back into an expanded state.

File: iOS/Flows/Conversation/Input/SwipeableInputAccessoryView.swift [hunk 3]
```diff
@@ -119,7 +119,7 @@ class SwipeableInputAccessoryView: BaseView {
                                          self.textViewCollapsedVerticalHeightContstraint])
             
             if !self.textView.text.isEmpty || self.textView.isFirstResponder {
-                textViewLeadingValue = self.addView.left + 40 
+                textViewLeadingValue = self.addView.left + AddMediaView.collapsedHeight
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Input/SwipeableInputAccessoryView.swift:122]: Trying to remove magic numbers.

File: iOS/Flows/Conversation/Input/SwipeableInputAccessoryView.swift [hunk 4]
```diff
@@ -135,9 +135,12 @@ class SwipeableInputAccessoryView: BaseView {
             if self.textView.numberOfLines > 2, let lineHeight = self.textView.font?.lineHeight {
                 let multiplier: CGFloat = clamp(CGFloat(self.textView.numberOfLines) - 2, 0, 2)
                 proposedHeight += lineHeight * multiplier
+                if proposedHeight > self.inputContainerHeightConstraint.constant - MessageContentView.textViewPadding {
+                    proposedHeight = (lineHeight * CGFloat(self.textView.numberOfLines)) + MessageContentView.textViewPadding
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Input/SwipeableInputAccessoryView.swift:139]: This helps ensure we have the proper padding.

File: iOS/Flows/Conversation/Input/SwipeableInputAccessoryView.swift [hunk 5]
```diff
@@ -162,7 +165,7 @@ class SwipeableInputAccessoryView: BaseView {
                 self.animationView.play(fromFrame: 20, toFrame: 30, loopMode: .playOnce, completion: nil)
             }
 
-            newAddViewSize = 100
+            newAddViewSize = self.addView.hasMedia ? AddMediaView.expandedHeight : AddMediaView.collapsedHeight
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Input/SwipeableInputAccessoryView.swift:168]: Don't show a large plus button when expanded.

File: iOS/Flows/Conversation/Input/InputHandlerCoordinator.swift (hunk 2 above)
Comment (MartinJibber) [reply iOS/Flows/Conversation/Input/InputHandlerCoordinator.swift:106]: Sure, let's try it out.

**Review by MartinJibber — APPROVED** [review]

**Pull request merged**